src/common/kube-helpers.ts

```diff
@@ -140,3 +141,23 @@ export function getNodeWarningConditions(node: V1Node) {
     c.status.toLowerCase() === "true" && c.type !== "Ready" && c.type !== "HostUpgrades"
   )
 }
+
+export function validateKubeConfig(config: KubeConfig) {
```

Comment:
using strings as both error reporting and success results is quite bad practice. Special strings are practically just as bad as special. I would very much recommend returning a `boolean` and doing a `throw new ExecValidationError(...)` here instead.

Reply:
@Nokel81 - do we have any existing examples of custom errors in Lens? I had a quick look and couldn't find one and want to make sure we can actually extend Error.

I agree though I would rather return an error rather than text. I may just try and see if I can get the error visible to user in my new function rather than in the AddCluster class.

Reply:
We don't (yet 😞) but I think that a validation function should only be concerned with reporting an error (if there is one) to its caller, not to the user. Since (in theory) we may want this to be called from several places.

I would do something like the following:

```typescript
class ExecValidationError extends Error {
    constructor(execPath: string) {
        super(`User Exec command "${execPath}" not found on host.`)
    }
}
```

and then in the caller of `validateKubeConfig` have a `try`/`catch` with something like the following:
```typescript
} catch (error) {
    if (error instanceOf ExecValidationError) {
        // notify user
    } else {
        throw error // rethrow errors we don't know how to handle
    }
}
```

Reply:
Awesome. Thank you @Nokel81.